File: app/src/main/java/com/blog/demo/application/storage/json/JsonData.java
package com.blog.demo.application.storage.json;

import com.blog.demo.People;

public class JsonData {
    public boolean aBoolean;
    public int aInt;
    public double aDouble;
    public String aString;
    public People aObject;
    public String[] aStringArray;
    public People[] aObjectArray;

    @Override
    public String toString() {
        return "aBoolean = " + aBoolean + ",\n"
                + "aInt = " + aInt + ",\n"
                + "aDouble = " + aDouble + ",\n"
                + "aString = " + aString + ",\n"
                + "aObject = " + aObject.description() + ",\n"
                + "aStringArray = " + toString(aStringArray) + ",\n"
                + "aObjectArray = " + toString(aObjectArray);
    }

    private String toString(String[] stringArray) {
        String string = "";
        for (String str : stringArray) {
            string += str + ",";
        }
        return string.substring(0, string.length() - 1);
    }

    private String toString(People[] objectArray) {
        String string = "";
        for (People people : objectArray) {
            string += people.description() + ",";
        }
        return string.substring(0, string.length() - 1);
    }

}

File: app/src/main/java/com/blog/demo/application/storage/json/JsonData.kt
package com.blog.demo.application.storage.json

import com.blog.demo.People

class JsonData {
    var aBoolean = false
    var aInt = 0
    var aDouble = 0.0
    var aString: String? = null
    var aObject: People? = null
    var aStringArray: Array<String?> = arrayOf()
    var aObjectArray: Array<People?> = arrayOf()

    override fun toString(): String {
        return """
             aBoolean = $aBoolean,
             aInt = $aInt,
             aDouble = $aDouble,
             aString = $aString,
             aObject = ${aObject!!.description()},
             aStringArray = ${stringArrayToString(aStringArray)},
             aObjectArray = ${objectArrayToString(aObjectArray)}
             """.trimIndent()
    }

    private fun stringArrayToString(stringArray: Array<String?>): String {
        var string = ""
        for (str in stringArray) {
            string += "$str,"
        }
        return string.substring(0, string.length - 1)
    }

    private fun objectArrayToString(objectArray: Array<People?>): String {
        var string = ""
        for (people in objectArray) {
            string += people?.description() + ","
        }
        return string.substring(0, string.length - 1)
    }

}